sdk/eventhub/event-hubs/src/sender.ts

```diff
@@ -58,6 +59,33 @@ export class EventHubProducer {
     this._eventHubSender = EventHubSender.create(this._context, partitionId);
   }
 
+  /**
+   * Creates a batch where event data objects can be added for later `Send` call
+   * @param options  Options to define partition key and max message size.
+   * @returns Promise<EventDataBatch>
+   */
+  async createBatch(options?: BatchOptions): Promise<EventDataBatch> {
+    if (!options) {
+      options = {};
+    }
+    let maxMessageSize = await this._eventHubSender!.getMaxMessageSize();
+    if (options.maxMessageSize) {
+      if (options.maxMessageSize > maxMessageSize) {
+        const error = new Error(
+          "Max message size is greater than maximum message size on the AMQP sender link."
+        );
+        log.error(
+          "[%s] Max message size is greater than maximum message size on the AMQP sender link. %O",
+          this._context.connectionId,
+          error
+        );
+        throw error;
+      }
+      maxMessageSize = options.maxMessageSize;
+    }
+    return new EventDataBatch(this._context, maxMessageSize, options.partitionKey!);
```

Comment:
options.partitionKey might be undefined, so we shouldn't add the `!` that indicates that it does exist.

Reply:
I guess @ShivangiReja needed the ! because partitionKey was mandatory in the constructor. Once it is made optional, we shouldnt need this